dpnp/dpnp_algo/dpnp_algo.pxd

```diff
@@ -205,12 +206,32 @@ cdef extern from "dpnp_iface.hpp":
 ctypedef void(*fptr_1out_t)(void *, size_t)
 ctypedef void(*fptr_1in_1out_t)(void * , void * , size_t)
 ctypedef void(*fptr_2in_1out_t)(void * , void*, void*, size_t)
+ctypedef void(*fptr_2in_1out_full_t)(const void *, const void *, const void *, const size_t, const size_t,
+                                     const long*, const long*, const size_t, const size_t)
 ctypedef void(*fptr_blas_gemm_2in_1out_t)(void * , void * , void * , size_t, size_t, size_t)
 ctypedef void(*dpnp_reduction_c_t)(void * , const void * , const size_t*, const size_t, const long*, const size_t, const void * , const long*)
 
 cdef dparray call_fptr_1out(DPNPFuncName fptr_name, result_shape, result_dtype)
 cdef dparray call_fptr_1in_1out(DPNPFuncName fptr_name, dparray x1, dparray_shape_type result_shape)
 cdef dparray call_fptr_2in_1out(DPNPFuncName fptr_name, dparray x1, dparray x2, dparray_shape_type result_shape)
+cdef dparray call_fptr_2in_1out_full(DPNPFuncName fptr_name, dparray x1, dparray x2)
+
+
+ctypedef fused dpnp_scalar:
+    int
+    long
+    float
+    double
+    bint
+    double complex
+
+ctypedef fused dpnp_input1:
+    dparray
+    dpnp_scalar
+
+ctypedef fused dpnp_input2:
+    dparray
+    dpnp_scalar
```

Comment:
avoid using fused types.
Looks like a workaround, please, redesign it.

Reply:
If you have redesign ideas let me know.

Reply:
fused types is just a feature for templates. You just don't really need it on cython layer.
And more over you are using this types on `dpnp_multiply` in bad way, where lost all cython templating idea.